auth/base/src/main/java/org/wildfly/security/auth/principal/NamePrincipal.java

```diff
@@ -93,4 +93,36 @@ public boolean equals(final NamePrincipal obj) {
     public String toString() {
         return name;
     }
+
+    /**
+     * Attempt to convert the given principal to a {@code NamePrincipal}.
+     *
+     * @param principal the original principal
+     * @return the {@code NamePrincipal} or {@code null} if the principal cannot be converted
+     */
+    public static NamePrincipal asNamePrincipal(Principal principal) {
```

Comment:
Just thinking about the naming conventions here - could the name be simplified to:

NamePrincipal from(Principal principal)

The calling code will be NamePrincipal.from(p) - if we wanted to expand our principal types supported maybe it could start a pattern of them having a from method to handle the conversion.

Reply:
Good point, will rename this to `from`.